granulate_utils/linux/elf.py

```diff
@@ -90,3 +97,52 @@ def is_statically_linked(path: str) -> bool:
             if segment.header.p_type == "PT_DYNAMIC":
                 return False
     return True
+
+
+def get_symbol_addr(path: str, sym_name: str) -> Optional[int]:
+    with open(path, "rb") as f:
+        elf = ELFFile(f)
+        symtab = elf.get_section_by_name(".symtab")
+        if symtab is None:
+            return None
+        symbols = symtab.get_symbol_by_name(sym_name)
+        if symbols is None or len(symbols) != 1:
```

Comment:
`len(symbols) != 1` should be considered an error state in this function, I think. Let's raise on it?

Reply:
You are absolutely correct, I copied it from `read_elf_symbol` (will be refactored to use same code after I add class or accept `Union`) - and I think it should be an error there too.